src/main/java/Duke.java

```diff
@@ -1,10 +1,333 @@
+
+    public void run() {
+        try {
+            File dataDirectory = storage.initialiseDirectory();
+            File dataFile = storage.initialiseFile(dataDirectory);
+
+            ui.greet();
+            storage.loadTasksFromFile(dataFile, tasks);
+
+            String commandLine;
+
+            do {
+                commandLine = ui.readCommand();
+                Command command = parser.parseInput(commandLine);
+
+                try {
+                    switch (command.getType()) {
+                    case "bye":
+                        ui.exit();
+                        break;
+                    case "list":
+                        displayTasksList(tasks);
+                        break;
+                    case "done":
+                        markDone(command, tasks);
+                        storage.saveTasksToFile(dataFile, tasks);
+                        break;
+                    case "delete":
+                        deleteTask(command, tasks);
+                        storage.saveTasksToFile(dataFile, tasks);
+                        break;
+                    case "todo":
+                        addToDoTask(command, tasks);
+                        storage.saveTasksToFile(dataFile, tasks);
+                        break;
+                    case "deadline":
+                        addDeadlineTask(command, tasks);
+                        storage.saveTasksToFile(dataFile, tasks);
+                        break;
+                    case "event":
+                        addEventTask(command, tasks);
+                        storage.saveTasksToFile(dataFile, tasks);
+                        break;
+                    case "print":
+                        printTasksOnDate(command, tasks);
+                        break;
+                    default:
+                        throw new DukeException("INVALID COMMAND. Please try again!");
+                    }
+                } catch (DukeException e) {
+                    ui.displayError(e.getMessage());
+                }
+            } while (!commandLine.equals("bye"));
+
+        } catch (DukeException | IOException e) {
+            ui.displayError(e.getMessage());
+        }
+    }
+
+    public void displayTasksList(TaskList tasks) throws DukeException {
+        if (tasks.size() != 0) {
+            String response = ui.tasksInYourList(tasks);
+            ui.displayResponse(response);
+        } else {
+            throw new DukeException("There are no tasks in your list!");
+        }
+    }
+
+    public void markDone(Command command, TaskList tasks) throws DukeException {
+        String commandDetails = command.getDetails();
+
+        if (commandDetails.trim().length() > 0) {
+            try {
+                int taskNum = Integer.parseInt(commandDetails);
+
+                if (taskNum >= 1 && taskNum <= tasks.size()) {
+                    tasks.get(taskNum - 1).setDone();
+                    Task taskDone = tasks.get(taskNum - 1);
```

Comment:
Not really related to coding standard but I think it might be better when you swap them around. Consider this:
``` java
Task taskDone = tasks.get(taskNum - 1);
taskDone.setDone();
```